### DISCUSSION TITLE
EcoFlow dynamic grid charging adjustment

### DISCUSSION THREAD

**Discussion by SweetHunter80** [svenerbe/ecoflow_dynamic_power_adjustment#4] [Ideas]
Hello,
Has anyone already found a way to charge the EcoFlow Delta Pro surplus via the household current?
Preferably combined with the script from Sven.
The script that Sven has created is TOP and I think there are other users here who have a 2nd PV that feeds into the house grid and it would be great to feed the excess energy into the battery instead of selling it cheaply, because the balcony power plant (PowerStream) feeds too little into the battery on bad days.
I know there will also be some people here who say it's better to sell the electricity and then buy it when you need it, which is also true that it doesn't necessarily pay off financially. But the environmental idea and the possibilities are more important to me.

**Comment by SweetHunter80** [comment]
![image](https://github.com/svenerbe/ecoflow_dynamic_power_adjustment/assets/129670092/f2dc6472-71f1-460c-a8b7-646930ce0ada)

**Comment by MhlTheOne** [comment]
You can't disable charging at all. There are the following two options:

1. Setting the charge level under battery soc ("params":{ "cmdSet": 32, "id": 49, "maxChgSoc": NEWVALUE })
2. Use an switchable power socket to disconnect powerstation from ac

**Comment by giovanne123** [comment]
I think you are searching for Delta Pro 🤔

So maybe only for information, the Delta 2 Max has a flag "chgPauseFlag" to pause AC load and to not always enabled/disable it at all.
I haven't searched to the docs if DP has something similiar... (https://developer-eu.ecoflow.com/us/document/deltapro)

**Reply by SweetHunter80** [reply]
Hi giovanne123
This is what I found, 

Setting the charge level | "params":{ "cmdSet": 32, "id": 49, "maxChgSoc": 100 } | "params":{ "cmdSet": 32, "id": 49,"quotas": ["ems.maxChargeSoc"] } | "data":{ "ems.maxChargeSoc": 0 }

Setting the discharge level | "params":{ "cmdSet": 32, "id": 51, "minDsgSoc": 10 } | "params":{ "cmdSet": 32, "id": 51,"quotas": ["ems.minDsgSoc"] } | "data":{ "ems.minDsgSoc": 0 }

but don't know how I could integrate it into the automation.
I assume that you have to write it in a script, which I can't do.
Is it possible to build a helper with it and then control the helper with an automation.

**Comment by giovanne123** [comment]
You have to enhance the script from Sven to support this too.

I don't use the script, was only for information here that there are maybe possibilities in the official API for the DP too !?

I have my own (quick and dirty) solution for my D2M & Co. by a python http (and mqtt) server serving my needed commands to the official API.

**Reply by SweetHunter80** [reply]
Ich benutze das Script von Sven.
Hab leider nicht das Wissen wie ich dieses erweitern könnte, da ich kein Python kann.
Danke für den Tipp.

**Reply by MhlTheOne** [reply]
Woher hast du die Werte der PowerStation aus dem Screenshot?

**Reply by MhlTheOne** [reply]
Nutzt du bereits die https://github.com/tolwi/hassio-ecoflow-cloud ?
Dann kannst du den Wert doch einfach per Automation setzen.

Beispiel (hier eine Delta2Max) per Curl:
```
curl \
  -H "Authorization: Bearer *******" \
  -H "Content-Type: application/json" \
  -d '{"entity_id": "number.delta2max_ac_charging_power", "value": 600}' \
  http://localhost:8123/api/services/number/set_value
```
Oder per Automation:
```
service: number.set_value
metadata: {}
data:
  value: "600"
target:
  entity_id: number.delta2max_ac_charging_power
```
Der Slider sollte dir auch bei der Pro zur Verfügung stehen.

**Reply by SweetHunter80** [reply]
Ja, hab die hassio-ecoflow-cloud.
Dann Probiere ich mal die Automatisation aus und Berichte.

**Reply by SweetHunter80** [reply]
Also Werte von 200 bis 2900 gehen. aber kein 0 also Pause.

**Reply by MhlTheOne** [reply]
Du kannst es nur so setzten wie es auch in der App möglich ist. Alternativ setzt du das battery level runter bis zu dem geladen wird. 
Letzte alternative wäre ein smart plug und damit die pro AC seitig trennen.
